### ISSUE TITLE
Crashing on Blender 3.0 Alpha

### ISSUE THREAD

**Issue by blessedcoolant** [JoseConseco/GoB#257]
When you try to import a mesh a SECOND time from zBrush to Blender, the addon crashes Blender.

The error is in GoB.py > run_import_periodically > bpy.ops.scene.gob_import(ctx) according to the crash log.

 Can be reproduced 100% of the time.

**Comment by kromar** [comment]
thanks for reporting the issue, i added a manual import option that should work as long as the automatic one will be solved in the latest master (v3, 5, 56)
Since 3.0 is still under heavy development it might be a blender but so i will have another look at this once 3.0 goes into beta phase.

**Comment by blessedcoolant** [comment]
Thank you for the update. I already did update to 3.5.55 and the manual option has been working well. I'll look forward to the automatic fix.

Sorry to bother you with this and this is more of a general query rather than to do with your addon, but I was wondering if you cold help me.

When I make changes and export from GoZ in zBrush, it always exports as a new mesh rather than updating the current mesh. So when I import in Blender using GoB, a new mesh is loaded rather than the currently existing mesh updating. Is there some option that lets me update the existing mesh rather than importing a whole new mesh?

**Comment by EdwHopper** [comment]
I confirm the crash on Import with Blender 3 alpha aswell that with Manual the import works only once.

**Comment by kromar** [comment]
can you try if this fixes the crashes for you?

v3.5.61
https://github.com/kromar/blender_GoB/tree/fix-import-crash

**Comment by blessedcoolant** [comment]
> rashes for you?

Nope still crashes.

**Comment by blessedcoolant** [comment]
> hm that is strange, can you tell me a bit more what you send over from zbrush? only one tool or multiple ones?
> what version of blender are you using?

Blender 3.0 Alpha. Sending just one tool.

**Comment by nameless667** [comment]
Hey Daniel! What happens in my case is that the transfer to blender works for the first time (it does not matter what the model is, it can be a single cube with 4 vertices) and the second one fails. In automatic import you get Blender crash with insta-desktop and in manual import nothing happens. In both cases the first transfer works as intended, second one fails. Blender 3.0 alpha (2.93 works perfectly)

**Comment by blessedcoolant** [comment]
@kromar Just wanted to report that the addon seems to be working with the latest build of 3.0 alpha. I am not facing crashes anymore.

One small thing though, when I send over a model from ZBrush, rather than updating the existing model in Blender, it is importing a new updated object. Am I missing a setting here or is the old functionality of updating the existing object broken?

**Comment by nameless667** [comment]
I can confirm that the latest 3.0 alpha version is crash free for me too(Gob 3.5.59). yay! @blessedcoolant probably a naming issue, on my end objects update like usual instead of having new instances.

**Comment by blessedcoolant** [comment]
> I can confirm that the latest 3.0 alpha version is crash free for me too(Gob 3.5.59). yay! @blessedcoolant probably a naming issue, on my end objects update like usual instead of having new instances.

When you say naming issue, do you think I should rename them to something other than what they were? Coz for me the name are the same .. for example: Cube gets imported as Cube.001 Cube.002 and etc indicating that the name is the same and hence the object names are being incremented by numbers. So not sure what to change on that end.

**Comment by nameless667** [comment]
@blessedcoolant Hmm, if the name is that simple, you are correct, it's weird. Does it keep happening if you rename to something else and try subsequent exports to Blender? I would clear the cache files just in case too on Zbrush side (Preferences->GoZ->Clear cache files)

**Comment by kromar** [comment]
i noticed some naming issues recently as well so i will need to dig a bit into that issue. Zbrush does some strange things when importing so maybe they changed some things in recent versions there that might cause some new issues.
Due to that it was required to rename certain suffix to something zbrush does not change on import. (for example cube.001 needs to be renamed to cube_001)
There is a new option in the addon preferences to export raw names from blender but it does not seem to work all the time for some reason.
i created a new issue for this naming topic to continue the discussion: https://github.com/JoseConseco/GoB/issues/267
